test/any/confluence/set_ops-test.es6.js

```diff
@@ -206,4 +231,158 @@ describe('Set ops', () => {
+      var nullSet = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+      }
+
+      set.orderBy(Num.ID).select(E.INTERSECT(nullSet))
+          .then(sink => {
+            expect(sink.a.map(num => num.id)).toEqual([]);
+            done();
+          });
+    });
+
+    it('should remove some items when secondary is subset', done => {
+      var set = MDAO.create({of: Num});
+      var secondary = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+        if ((i % 2) === 0) {
+          secondary.put(mkNum(i));
+        }
+      }
+
+      // {0..9} INTERSECT {0, 2, 4, 6, 8}
+      set.orderBy(Num.ID).select(E.INTERSECT(secondary))
+          .then(sink => {
+            expect(sink.a.map(num => num.id))
+                .toEqual([0, 2, 4, 6, 8]);
+
+            done();
+          });
+    });
+
+    it('should remove some items when primary is subset', done => {
+      var set = MDAO.create({of: Num});
+      var secondary = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        if ((i % 2) === 0) {
+          set.put(mkNum(i));
+        secondary.put(mkNum(i));
+        }
+      }
+
+      // {0, 2, 4, 6, 8} INTERSECT {0..9}
+      set.orderBy(Num.ID).select(E.INTERSECT(secondary))
+          .then(sink => {
+            expect(sink.a.map(num => num.id))
+                .toEqual([0, 2, 4, 6, 8]);
+
+            done();
+          });
+    });
+
+    it('should support composition', done => {
+      var set = MDAO.create({of: Num});
+      var secondary1 = MDAO.create({of: Num});
+      var secondary2 = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+      }
+
+      secondary1.put(mkNum(1));
+      secondary1.put(mkNum(7));
+
+      secondary2.put(mkNum(3));
+      secondary2.put(mkNum(7));
+      secondary2.put(mkNum(5));
+
+      // {0..9} INTERSECT {1, 7} INTERSECT {3, 7, 5}
+      set.orderBy(Num.ID)
+          .select(E.INTERSECT(secondary1, E.INTERSECT(secondary2)))
+          .then(sink => {
+            expect(sink.a.map(num => num.id))
+                .toEqual([7]);
+
+            done();
+          });
+    });
+
+    it('should deliver items in order, even when secondary would resolve find()s out of order', done => {
+      var set = MDAO.create({of: Num});
+      var secondary = EvilDAO.create({
+        of: Num,
+        delegate: MDAO.create({of: Num}),
+      });
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+        if ((i % 2) === 0) {
+          secondary.put(mkNum(i));
+        }
+      }
+
+      // {0..9} INTERSECT {0, 2, 4, 6, 8}
+      set.orderBy(Num.ID).select(E.INTERSECT(secondary))
+          .then(sink => {
+            expect(sink.a.map(num => num.id))
+                .toEqual([0, 2, 4, 6, 8]);
+
+            done();
+          });
+    });
+
+    it('should reject when primary rejects select()', done => {
```

Comment:
Since this is a property of the parent class, it's probably not providing much value to test it on each subclass.